smart_contract/contracts/test/Faucet.t.sol

```diff
@@ -0,0 +1,182 @@
+// SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+/**
+ * @title Faucet - Test
+ */
+
+import { CheatCodes } from 'foundry-test-utility/contracts/utils/cheatcodes.sol';
+import { Constants } from "./utils/constants.t.sol";
+
+import { Faucet } from "../Faucet.sol";
+
+contract FaucetTest is Constants {
+    Faucet private faucet;
+
+    function setUp() public {
+        vm.startPrank(ADMIN);
+
+        // Deploy contracts
+        faucet = new Faucet();
+
+        faucet.addMinter(address(this));
+
+        vm.deal(address(faucet), BASIC_MINT_AMOUNT);
+
+        vm.stopPrank();
+        vm.warp(1 hours);
+    }
+    
+    function test_Native_mint(
+        address to_
+    ) public {
+        vm.assume(to_ != address(0));
+        vm.assume(to_.code.length == 0);
+        uint256 currentBalance = address(to_).balance;
+        uint256 expectedBalance = currentBalance + BASIC_MINT_AMOUNT;
+
+        assertEq(address(to_).balance, currentBalance);
+        assertEq(address(faucet).balance, BASIC_MINT_AMOUNT);
+
+        faucet.requestTokens(to_);
+
+        assertEq(address(to_).balance, expectedBalance);
+        assertEq(address(faucet).balance, 0);
+    }
+
+    function test_requestTokenBeforeLockTime(
+        address to_
+    ) public {
+        vm.assume(to_ != address(0));
+
+        faucet.requestTokens(to_);
+
+        vm.warp(block.timestamp + 1);
+        verify_revertCall(RevertStatus.Faucet_InsufficientTimeElapsed);
```

Comment:
This test is failing

Reply:
Yes I was missing some assumption to make sure the fuzzing of the receiver does not fall on address(faucet) or address(1) :man_facepalming: